### ISSUE TITLE
Upstream - Unable to build YYC due to compiler error

### ISSUE THREAD

**Issue by hagusen** [offalynne/Input#371]
errors from __input_config_icons.
if removed then build works fine 
[it doesn't like the spaces in the names]

![image](https://user-images.githubusercontent.com/39492701/173554343-bed5a4a1-a800-4d20-8410-03d88150d1fa.png)

**Comment by JujuAdams** [comment]
GameMaker truly is a piece of shit.

**Comment by JujuAdams** [comment]
No clean workaround is available for this. I have reported this issue to YYG.

**Comment by JulianDicken** [comment]
New PR isnt breaking, hope you can use it @hagusen

**Comment by offalynne** [comment]
think a suitable workaround would be to use the ‘finalize’ pattern with the INPUT_ICONS macro to do array -> struct, avoiding the stupid underscores/spaces thing.

yyc doesn’t choke on spaces in keys while running, only fails to compile when they are directly declared, so

`[ ”xbox 360”, [[ “start”, “start”], […]`
`{ ”xbox 360” : { “start”: “start” , […]`

🤔 @JujuAdams ?

**Comment by JujuAdams** [comment]
Alternative:
```
input_icons("xbox 360")
.add("gamepad face south", "A")
.add("gamepad face east",  "B")
.add("gamepad face west",  "X")
.add("gamepad face north", "Y")
```

**Comment by offalynne** [comment]
fluuuent. any reason to prefer the same format for keynames and blocked keys, on the subject ?

**Comment by JujuAdams** [comment]
Fluent interface implemented for in 5.1

Would like to stick with what we know works already for key names and blocked keys for the sake of minimising risk.
